src/App.tsx

```diff
@@ -115,11 +125,16 @@ function App(): ReactElement {
    * render. Also update other UI elements, including the plot and URL.
    */
   useEffect(() => {
+    // TODO: This is getting larger... ColorizeCanvas needs to be refactored to take
+    // all of these in as props.
```

Comment:
Seeing this makes me think of website-3d-cell-viewer, where these prop updates each have their own effect so that one change doesn't call every update method... But several of volume-viewer's props are expensive to update, and pretty much all of the updaters below make cheap updates to shader uniforms. Sounds like this would be a convenient and appropriate pattern here, definitely worth keeping in mind (or in a TODO) for the future!